pkg/platform/architecture_solaris.go

```diff
@@ -0,0 +1,16 @@
+package platform
+
+import (
+	"os/exec"
+	"strings"
+)
+
+// runtimeArchitecture get the name of the current architecture (i86pc, sun4v)
+func runtimeArchitecture() (string, error) {
+	cmd := exec.Command("/usr/bin/uname", "-m")
+	machine, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(machine)), nil
```

Comment:
hum, `TrimSpace` is not done on `architecture_freebsd.go`, any reason for that ? If it would work the same for both, maybe a `architecture_unix.go` would be better for both of them.